### DISCUSSION TITLE
Unable to get correctly params from api route

### DISCUSSION THREAD

**Discussion by lalainarahajason** [vercel/next.js#57977] [Help]
### Summary

Hi,

Just wonder how to get the id from this snippet ? : 

folders structure : 
- app
--- api
--- events
-----[id]
-------route.js

`const handler = async (req, res) => { console.log(req.query} } 
export default {handler as GET, handler as POST}`

When doing : mysite.com/api/events/12 , the console print {} instead of desired data

Why ?

Thank you

**Comment by icyJoseph** [comment]
I think you want to use:

- https://nextjs.org/docs/app/building-your-application/routing/route-handlers#dynamic-route-segments

There's no `res` passed as second argument in Router Handlers, but there you'll get a params object, from where you could read 
the `id`

```ts
export async function GET(
  request: Request,
  { params }: { params: { slug: string } }
) {
  const slug = params.slug // 'a', 'b', or 'c'
}
```

From pages API Routers, to app Router Handlers, the API has changed. Please do go through the docs: https://nextjs.org/docs/app/building-your-application/routing/route-handlers#url-query-parameters

**Reply by lalainarahajason** [reply]
Thank you, i'll try it
